### ISSUE TITLE
Create Manage Donation Form Page

### ISSUE THREAD

**Issue by jrock-3** [ctc-uci/patriots-and-paws-frontend#41]
### Backend
- Create a new table `furniture_options` to store all furniture items that PNP accepts. 
  - Add CRUD routes (GET all options, POST a new option(s), DELETE an option)
- Update `furniture` table to include a `count`. Return that count in all POST requests, and add that data to the database in the GET requests.
  - [NOTE: This is blocked by a previous sprint's task, don't make the changes yet]

### Frontend
- Create `EditItemsList` component
- Create `ItemCard` component
- Create `ManageDonationFormPage` component

## Notes
- The `furniture` table is not directly related to this task; we just needed it updated for the future
- Furniture option items are sorted into `Items We Accept` and `Items We DO NOT Accept` by the `accepted` column in the `furniture_options` table
### Edit Items List
- [NOTE: Validation implementation approach TBD so don't implement it yet]
- Text input section to input furniture option
  - Add validation to ensure the item trying to be added is alphanumeric
- `Add Item to List` button to add the furniture option to the item list
  - Add validation to ensure the item trying to be added isn't already in the `furniture_options` table
- Display current items list in a section with a list of `ItemCard` components
### Item Card
- Use Chakra `Card` component
- Display furniture option item name
- `Close Button` to delete furniture option
  - Open an `Alert Dialogue` to confirm deletion
### Manage Donation Form Page
- Have two `EditItemsList` components side by side, one for `Items We Accept` and other for `Items We DO NOT Accept`
- Add `Publish` button which adds all of the new options to the `furniture_options` table
  - The `furniture_options` `put` request should only be made after clicking `Publish`
- Add `Preview` button which shows the `ItemInfo` modal (this component is implemented in another task, just add the capability of opening a modal to this button)

## Acceptance Criteria
- New table is created and the `furniture` table is updated.
- You can add a new item name to the list. Adding it automatically populates the `EditItemsList` component
- A duplicate name should error when entering the name in the `AddItemModal`
- You can delete an item from the list.
- Publishing the form should make a backend request and update the database.

## Resources
- [Manage Donation Form Figma](https://www.figma.com/file/3DOUy8M8f1aWmKbpC63XMC/PNP-Wireframes?node-id=1353%3A14053&t=QXAtxQA0oRic9IMY-0)
- [Database model](https://www.figma.com/file/eFv2YZ1ARepQrDWI3lTcyd/Database-Model)
- [Chakra Card](https://chakra-ui.com/docs/components/card)
- [Chakra Modal](https://chakra-ui.com/docs/components/modal)
- [Corresponding backend issue](https://github.com/ctc-uci/patriots-and-paws-backend/issues/24) - use this to create your pr.
